full-service/src/service/txo.rs

```diff
@@ -123,19 +123,15 @@ where
         fee: Option<String>,
         tombstone_block: Option<String>,
     ) -> Result<TxProposal, TxoServiceError> {
-        use crate::service::txo::TxoServiceError::{TxoNotSpendable, TxoNotSpendableByAnyAccount};
+        use crate::service::txo::TxoServiceError::TxoNotSpendableByAnyAccount;
 
         let conn = self.wallet_db.get_conn()?;
         conn.transaction(|| {
             let txo_details = Txo::get(&txo_id.to_string(), &conn)?;
-            let received_to_account_txo_status = match txo_details.received_to_account {
-                Some(account_status) => Ok(account_status),
-                None => Err(TxoNotSpendableByAnyAccount(txo_details.txo.txo_id_hex)),
-            }?;
 
-            let account_id_hex = match received_to_account_txo_status.txo_status.as_str() {
-                TXO_STATUS_UNSPENT => Ok(received_to_account_txo_status.account_id_hex),
-                _ => Err(TxoNotSpendable(received_to_account_txo_status.txo_status)),
+            let account_id_hex = match txo_details.received_account_id_hex {
```

Comment:
I believe this can be written as `let account_id_hex = txo_details.received_account_id_hex.ok_or_else(|| TxoNotSpendableByAnyAccount(txo_details.txo_id_hex))?;`

Reply:
Good catch.